### PR TITLE
Fix make target cluster-sync-cdi, add cluster-clean-cdi & cluster-clean-test-infra

### LINKED ISSUE #1496: make cluster-sync-cdi deletes test pods
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind enhancement

**What happened**:
If you run `make cluster-sync-cdi`, it depends on the `make cluster-clean` target which deletes the entire cdi namespace, including test pods.
This makes the target less useful: it means that `make test-functional` cannot run.

### PR DESCRIPTION
**Pull request by arnongilboa**
Signed-off-by: Arnon Gilboa <agilboa@redhat.com>

**What this PR does / why we need it**:
`make cluster-sync-cdi` now depends on the new `cluster-clean-cdi` target which does not deletes the entire cdi namespace, and keeps the test pods running. Also, `make cluster-sync-test-infra` now depends on the new `cluster-clean-test-infra` target.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1496 

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by brybacki** [comment]
it would be nice to also update /hack/README.md#getting-started-for-developers

**Comment by brybacki** [comment]
comparing the Makefile to this readme I can tell it is really outdated: hack/README.md

**Comment by arnongilboa** [comment]
@brybacki @maya-r please review the changes in README.md.

File: hack/README.md [hunk 1]
```diff
@@ -53,14 +53,18 @@ The standard workflow is performed inside a helper container to normalize the bu
...
 - `cluster-down`: stop the cluster, doing a make cluster-down && make cluster-up will basically restart the cluster into an empty fresh state.
 - `cluster-down-purge`: cluster-down and cleanup all cached images from docker registry. Accepts [make variables](#make-variables) DOCKER_PREFIX. Removes all images of the specified repository. If not specified removes localhost repository of current cluster instance.
-- `cluster-sync`: builds the controller/importer/cloner, and pushes it into a running cluster. The cluster must be up before running a cluster sync. Also generates a manifest and applies it to the running cluster after pushing the images to it.
+- `cluster-sync`: 'make cluster-sync-cdi' followed by 'make cluster-sync-test-infra'
+- `cluster-sync-cdi`: builds the controller/importer/cloner, and pushes it into a running cluster. The cluster must be up before running a cluster sync. Also generates a manifest and applies it to the running cluster after pushing the images to it.
     - `cluster-sync-controller`: builds the controller and pushes it into a running cluster.
```
Comment (maya-r) [inline hack/README.md:58]: these aren't targets, right? so we can just remove these lines.
Happy to just do this in a separate PR though.

**Comment by maya-r** [comment]
/lgtm
we can always do the cleanup in another PR

File: hack/README.md (hunk 1 above)
Comment (arnongilboa) [reply hack/README.md:58]: which ones are not targets?

Comment (maya-r) [reply hack/README.md:58]: cluster-sync-importer, cluster-sync-cloner, etc.

Comment (arnongilboa) [reply hack/README.md:58]: please do it in separate PR, as I see there are many other dead targets as well.

Comment (brybacki) [reply hack/README.md:58]: yes, many more :), and true, it can be done in other PR
